javascriptv3/example_code/ssm/scenarios/ssm-basics.js

```diff
@@ -0,0 +1,447 @@
+  "sdkCreateSSMDoc",
+  /**
+   * @param {{ documentName: string }} c
+   */
+  async (c) => {
+    const contentData = `{
+                "schemaVersion": "2.2",
+                "description": "Run a simple shell command",
+                "mainSteps": [
+                    {
+                        "action": "aws:runShellScript",
+                        "name": "runEchoCommand",
+                        "inputs": {
+                          "runCommand": [
+                            "echo 'Hello, world!'"
+                          ]
+                        }
+                    }
+                ]
+            }`;
+    try {
+      const _response = await client.send(
+        new CreateDocumentCommand({
+          Content: contentData,
+          Name: c.documentName,
+          DocumentType: "Command",
+        })
+      );
+    } catch (e) {
+      console.log("Exception type: (" + typeof e + ")");
+      if (e instanceof DocumentAlreadyExists)
+        console.log("Document already exists. Continuing...\n");
+      else throw e;
+    }
+  }
+);
+
+const ec2HelloWorld = new ScenarioOutput(
+  "ec2HelloWorld",
+  `Now you have the option of running a command on an EC2 instance that echoes 'Hello, world!'. In order to run this command, you must provide the instance ID of a Linux EC2 instance. If you do not already have a running Linux EC2 instance in your account, you can create one using the AWS console. For information about creating an EC2 instance, see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-launch-instance-wizard.html.`
+);
+
+const confirmEC2HelloWorld = new ScenarioInput(
+  "confirmEC2HelloWorld",
+  "Enter your EC2 InstanceId or press enter to skip this step: ",
+  { type: "input", default: "" }
+);
+
+const sdkEC2HelloWorld = new ScenarioAction(
+  "sdkEC2HelloWorld",
+  /**
+   * @param {{ documentName: string, confirmEC2HelloWorld: string }} c
+   */
+  async (c) => {
+    const response = await client.send(
+      new SendCommandCommand({
+        DocumentName: c.documentName,
+        InstanceIds: [c.confirmEC2HelloWorld],
+      })
+    );
+    c.CommandId = response.Command.CommandId;
+  },
+  {
+    skipWhen: /** @param {{ confirmEC2HelloWorld: string }} c */ (c) =>
+      c.confirmEC2HelloWorld === "",
+  }
+);
+
+const getCommandTime = new ScenarioOutput(
+  "getCommandTime",
+  "Lets get the time when the specific command was sent to the specific managed node."
+);
+
+const sdkGetCommandTime = new ScenarioAction(
+  "sdkGetCommandTime",
+  /**
+   * @param {{ requestedDateTime: Date }} c
+   */
+  async (c) => {
+    let listInvocationsPaginated = [];
+    for await (const page of paginateListCommandInvocations(
+      { client },
+      { CommandId: c.CommandId }
+    )) {
+      listInvocationsPaginated.push(...page.CommandInvocations);
+    }
+    /**
+     * @type {import('@aws-sdk/client-ssm').CommandInvocation}
+     */
+    const commandInvocation = listInvocationsPaginated.shift();
```

Comment:
We send the Id of the command, so only one should return, so grabbing the first (only) result makes sense.

I guess the better question is if this call still needs to be paginated? I did it that way because we generally say paginate any calls which can be, but maybe that doesn't make sense in this case.

Reply:
Pagination still makes sense. This is how you would do it. I'm thinking just a brief comment explaining what you just told me.